core/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -301,6 +305,21 @@ public OpenTelemetryRumBuilder addLogRecordExporterCustomizer(
         return this;
     }
 
+    /**
+     * Enables a ratio-based probabilistic sampling of sessions. When a session is sampled
+     * (included), then the telemetry for that session will be created and exported. If a session is
+     * not sampled (excluded) then the telemetry for that session will not be exported.
+     *
+     * @param ratio - A number between 0 and 1, representing a percentage of sessions to sample.
+     */
+    public OpenTelemetryRumBuilder enableSessionBasedSampling(double ratio) {
+        if (ratio < 0.0 || ratio > 1.0) {
+            throw new IllegalArgumentException("ratio must be in range [0.0, 1.0]");
```

Comment:
should we log out and default to 0.0 instead? I usually don't like to throw exceptions because maybe a simple mistake and a wrong release might crashpeople'ss apps, and a new update alwaytakeske days

Reply:
Yeah that's reasonable. I think I agree. 👍🏻